src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -126,13 +126,42 @@ async function buildTxProposal(req, res) {
     }
 
     let createdProposalId;
-    if (constants.SWAP_SERVICE_FEATURE_TOGGLE && serviceParams.is_new) {
-      const { proposalId } = await atomicSwapService.serviceCreate(
-        req.walletId,
-        proposal.partialTx.serialize(),
-        serviceParams.password
-      );
-      createdProposalId = proposalId;
+    if (constants.SWAP_SERVICE_FEATURE_TOGGLE) {
+      if (serviceParams.is_new) {
+        // Handling the creation of a new proposal with the Atomic Swap Service
+        const { proposalId } = await atomicSwapService.serviceCreate(
+          req.walletId,
+          proposal.partialTx.serialize(),
+          serviceParams.password
+        );
+        createdProposalId = proposalId;
+      } else if (serviceParams.proposal_id) {
+        // Handling the update of an existing proposal with the Atomic Swap Service
+
+        // First, validate if the proposal is already registered with this wallet
+        const listenedProposals = await atomicSwapService.getListenedProposals(req.walletId);
+        if (!listenedProposals.has(serviceParams.proposal_id)) {
+          res.status(404);
+          res.send({
+            success: false,
+            error: 'Proposal is not registered. Register it first through [POST] /register/:proposalId',
+          });
+          return;
+        }
+
+        // Retrieving registered proposal password
+        const requestedProposal = listenedProposals.get(serviceParams.proposal_id);
+        const { password } = requestedProposal;
+
+        await atomicSwapService.serviceUpdate(
```

Comment:
We do not check for the success response, should we?

Reply:
The service already checks for the success and throws if it isn't `true`, so we don't need to double check here.